### ISSUE TITLE
Rails 5.2.1 with ransack 1.8.9

### ISSUE THREAD

**Issue by kennyadsl** [activerecord-hackery/ransack#943]
Hey there 👋, 

[The latest version (1.8.9) released today in rubygems](https://rubygems.org/gems/ransack/versions/1.8.9) locks Rails compatibility up to 5.1.1. I saw that these are changes done in the branch [1-8-stable](https://github.com/activerecord-hackery/ransack/tree/1-8-stable) only and not in master. Is releasing the gem this way intentional?

I think that the issue with this approach is that other gems that uses Ransack as a dependency in their gemspec files cannot use the GitHub `master` branch version of this gem as suggested [here](https://github.com/activerecord-hackery/ransack/issues/942#issuecomment-411478792) and this is letting impossible for them to support Rails > 5.2.0.

Thanks!

**Comment by gregmolnar** [comment]
I am not sure how do you mean other gems using ransack as a dependency, but the reason I released that version is to stop people filing issues about Rails 5.2.1 incompatibility. Hopefully now they get the updated version and it drives them here so they can see master branch is available for Rails 5.2.1 till I release version 2 of Ransack(probably this weekend).

**Comment by kennyadsl** [comment]
I meant that ransack can be required as dependency [like we do in Solidus](https://github.com/solidusio/solidus/blob/425240b84f670558aab61df2655be9a75e00dda0/core/solidus_core.gemspec#L41). I think we can't require the GitHub `master` branch there, right?

I think in the meantime we'll fix our issue locking rails dependency to 5.2.0 and using ransack 1.8.8, so take your time! 

Thanks for your answer and your work on ransack.

**Comment by gregmolnar** [comment]
I see what you mean now. In situations like that I use a Gemfile, which embeds the gemspec and require dependencies from github in the gemfile. 
```ruby
# Gemfile
source 'https://rubygems.org'
gemspec

gem "ransack", github: "activerecord-hackery/ransack"
```

**Comment by vcsjones** [comment]
According to bundler, Ransack requires `<= 5.1.1` for actionpack:

<img width="498" alt="image" src="https://user-images.githubusercontent.com/361677/43903244-fcfe273e-9bb9-11e8-8f77-c0de0f35ca4e.png">

The rubygem's site agrees for 1.8.9:

<img width="350" alt="image" src="https://user-images.githubusercontent.com/361677/43903268-0f4b34e0-9bba-11e8-890a-0c6a7e75d051.png">

So ransack does not appear to work for Rails 5.2.1, only 5.1.1 or below.

If I look at the previous version, 1.8.8, it has no upper-limit on the actionpack version:

<img width="274" alt="image" src="https://user-images.githubusercontent.com/361677/43903323-344c1052-9bba-11e8-8229-c38445d6a8c5.png">

**Comment by gregmolnar** [comment]
@vcsjones For Rails >=5.2 use the master branch till the next release, because 1.8.x versions will never be compatible with newer versions of Rails.

**Comment by vcsjones** [comment]
@gregmolnar So even version 1.8.8 which did not have a restriction on the upper version of activerecord and actionpack technically doesn't work with Rails 5.2?

**Comment by gregmolnar** [comment]
Seriously, how would that work with the newest release of Rails if the one after that doesn't even work with it?

**Comment by vcsjones** [comment]
Sorry if I was not clear, I mean to say that we have been using 1.8.8 with Rails 5.2.0 without issue. So even if we stay at Rails 5.2.0 for Rails, we cannot upgrade to 1.8.9 because it introduces a requirement of Rails 5.1.1 or lower.

**Comment by gregmolnar** [comment]
Oh, I get it. Well, 1.8.8 and 1.8.9 is pretty much the same, the main difference is I locked support for Rails in the later. The reason to lock it at 5.2.0 is, because there are bugfixes on master which will be released as 2.0.0 and I want to encourage everybody over 5.2 to use that version, not the 1.8.x line.
For now, if you are staying on 5.2.0, you can either use 1.8.8, or point to the master branch till I release the next version(this weekend).

**Comment by gregmolnar** [comment]
2.0.0 with Active Record 5.2.1 support has been released. Please give it a spin and let us know if you encounter any issues.
